### DISCUSSION TITLE
Use more than one disk type for fs.configure

### DISCUSSION THREAD

**Discussion by SystemZ** [seaweedfs/seaweedfs#5360] [Q&A]
Hi, is there a way to specify that volume or EC shard can use any available disk type or more than one in fs.configure?
Currently not specifing disk type creates volumes on volume servers with not specified disk type only.

I'm using small cluster with 4 volume servers.
2 volume servers are launched without disk type specified so it's default disk or "".
Previously I marked them as disk type "ssd" but there were problems in cluster.check that demanded default type empty or hdd and cluster logs were stuck and not writing with a wall of errors.
Remaining 2 volume servers are marked as disk type "nvme".

Most of data I store on cluster doesn't need any redundancy.
It's data that can be downloaded again from other location or generated.
Remaining, small amount of data in few buckets are just very valuable and I want to copy them on all my volume servers as volume or as EC shards to be available even when all disks but one fails.

If I would specify all disks (nvme and ssd) as ssd, nvme or "" on volume servers, then I would loose ability to direct writes for new data to exclusively NVMe disks.
Is there a way to have both of those features at same time?

**Comment by chrislusf** [comment]
One volume server can have multiple directories. Another way is to add a directory with a different disk type.

**Reply by SystemZ** [reply]
That would complicate a bit more managing of physical disks.
I assume creating new disk type like `ssd-plus-nvme` and using new folder for volume server with that disk type would help.
Doesn't seems to be fancy solution but it would work, thanks! :D

I'll wait for more ideas but I really appreciate such fast response from maintainer himself ❤️
BTW I really like idea and execution of SeaweedFS, big thanks @chrislusf 🙌
